module_utils/network_lsr/argument_validator.py

```diff
@@ -271,8 +271,12 @@ def __init__(  # pylint: disable=too-many-arguments
     def _validate_impl(self, value, name):
         v = None
         try:
-            if isinstance(value, self.numeric_type):
-                v = value
+            if self.numeric_type is not bool and isinstance(value, bool):
```

Comment:
The class is called `ArgValidatorNum`, the `self.numeric_type is not bool`  should always true.
Did I miss anything?

Reply:
`ArgValidatorNum` is an API, where the user can specify the numeric type. You are correct, that no user currently sets this to bool, so the condition is never satisfied.

However, implementing it also serves as a form of documentation for the fact that `bool` in python is a subtype of `int`, which requires some special considerations here. And, beside "documentation",  it just does the right thing in those cases too.

But yes, it could be dropped too!

Reply:
`self.numeric_type is not bool` was dropped.